pkg/cosign/keys_test.go

```diff
@@ -48,7 +130,66 @@ func TestLoadECDSAPrivateKey(t *testing.T) {
 	if _, err := rand.Read(buf[:]); err != nil {
 		t.Fatal(err)
 	}
-	if _, err := LoadECDSAPrivateKey(buf[:], []byte("wrong")); err == nil {
+	if _, err := LoadPrivateKey(buf[:], []byte("wrong")); err == nil {
 		t.Error("expected error decrypting key!")
 	}
 }
+
+func TestImportPrivateKey(t *testing.T) {
+	td := t.TempDir()
+	err := os.WriteFile(filepath.Join(td, "validrsa.key"), []byte(validrsa), 0600)
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	// Test valid RSA private key
+	if _, err = ImportKeyPair(filepath.Join(td, "validrsa.key"), pass("hello")); err != nil {
+		t.Errorf("unexpected error importing key: %s", err)
+	}
+	os.Remove(filepath.Join(td, "validrsa.key"))
+
+	err = os.WriteFile(filepath.Join(td, "invalidrsa.key"), []byte(invalidrsa), 0600)
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	// Test invalid RSA private key
+	if _, err = ImportKeyPair(filepath.Join(td, "invalidrsa.key"), pass("hello")); err == nil {
+		t.Errorf("unexpected error importing key: %s", err)
+	}
+	os.Remove(filepath.Join(td, "invalidrsa.key"))
+
+	err = os.WriteFile(filepath.Join(td, "invalidkey.key"), []byte(invalidkey), 0600)
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	// Test invalid PGP private key
+	if _, err = ImportKeyPair(filepath.Join(td, "invalidkey.key"), pass("hello")); err == nil {
+		t.Errorf("unexpected error importing key: %s", err)
+	}
+	os.Remove(filepath.Join(td, "invalidkey.key"))
+
+	err = os.WriteFile(filepath.Join(td, "validec.key"), []byte(validec), 0600)
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	// Test valid EC private key
+	if _, err = ImportKeyPair(filepath.Join(td, "validec.key"), pass("hello")); err != nil {
```

Comment:
consider defining these files above f := filepath.Join(td, "validec.key") in these places so you can re-use it.